dace/transformation/dataflow/redundant_array.py

```diff
@@ -97,4 +97,107 @@ def gnode(nname):
             RedundantArray._arrays_removed += 1
 
 
+class RedundantArrayInverse(pm.Transformation):
+    """ Implements the redundant array removal transformation, applied
+        when a transient array is copied from and to (from another array),
+        but never used anywhere else. This transformation remove the second
+        array. """
+
+    _arrays_removed = 0
+    _in_array = nodes.AccessNode("_")
+    _out_array = nodes.AccessNode("_")
+
+    @staticmethod
+    def expressions():
+        return [
+            nxutil.node_path_graph(RedundantArray._in_array,
+                                   RedundantArray._out_array)
+        ]
+
+    @staticmethod
+    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+        in_array = graph.nodes()[candidate[RedundantArray._in_array]]
+        out_array = graph.nodes()[candidate[RedundantArray._out_array]]
+
+        # Ensure in degree is one (only one source, which is in_array)
+        if graph.in_degree(out_array) != 1:
+            return False
+
+        # Make sure that the candidate is a transient variable
+        if not out_array.desc(sdfg).transient:
+            return False
+
+        # Make sure that both arrays are using the same storage location
+        if in_array.desc(sdfg).storage != out_array.desc(sdfg).storage:
+            return False
+
+        # Find occurrences in this and other states
+        occurrences = []
+        for state in sdfg.nodes():
+            occurrences.extend([
+                n for n in state.nodes() if isinstance(n, nodes.AccessNode)
+                and n.desc(sdfg) == out_array.desc(sdfg)
+            ])
+
+        if len(occurrences) > 1:
+            return False
+
+        # Only apply if arrays are of same shape (no need to modify memlet subset)
+        # if len(in_array.desc(sdfg).shape) != len(
+        #         out_array.desc(sdfg).shape) or any(i != o for i, o in zip(
+        #             in_array.desc(sdfg).shape,
+        #             out_array.desc(sdfg).shape)):
+        #     return False
+
+        return True
+
+    @staticmethod
+    def match_to_str(graph, candidate):
+        out_array = graph.nodes()[candidate[RedundantArray._out_array]]
+
+        return "Remove " + str(out_array)
+
+    def apply(self, sdfg):
+        def gnode(nname):
+            return graph.nodes()[self.subgraph[nname]]
+
+        graph = sdfg.nodes()[self.state_id]
+        in_array = gnode(RedundantArray._in_array)
+        out_array = gnode(RedundantArray._out_array)
```

Comment:
Static variables must use the correct class here (too).

Reply:
Fixed